fdbserver/TenantBalancer.actor.cpp

```diff
@@ -1691,25 +1825,27 @@ ACTOR Future<Void> clearTenant(TenantBalancer* self,
 	if (doErase) {
 		// Erase the moved data, if desired
 		KeyRange rangeToErase = prefixRange(targetPrefix);
-		wait(waitOrError(runTenantBalancerTransaction<Void>(self->db,
-		                                                    self->tbi.id(),
-		                                                    "ClearTenantErase",
-		                                                    [rangeToErase](Reference<ReadYourWritesTransaction> tr) {
-			                                                    tr->clear(rangeToErase);
-			                                                    return tr->commit();
-		                                                    }),
-		                 record->onAbort()));
-
-		TraceEvent("TenantBalancerPrefixErased", self->tbi.id())
+		Future<Void> clearTenantOrAbort =
+		    waitOrError(runTenantBalancerTransaction<Void>(self->db,
+		                                                   self->tbi.id(),
+		                                                   "ClearTenantErase",
+		                                                   [rangeToErase](Reference<ReadYourWritesTransaction> tr) {
+			                                                   tr->clear(rangeToErase);
+			                                                   return tr->commit();
+		                                                   }),
+		                record->onAbort());
+		wait(timeoutError(clearTenantOrAbort, SERVER_KNOBS->TENANT_BALANCER_OPERATION_TIMEOUT));
+		TraceEvent(SevDebug, "TenantBalancerPrefixErased", self->tbi.id())
```

Comment:
I think for now its probably good for us to have a log history of erasing prefixes at info level.

```suggestion
		TraceEvent("TenantBalancerPrefixErased", self->tbi.id())
```